okhttp-android-support/src/main/java/com/squareup/okhttp/internal/huc/JavaApiConverter.java

```diff
@@ -107,15 +117,72 @@ public static Response createOkResponse(URI uri, URLConnection urlConnection) th
   }
 
   /**
+   * Returns headers for the header names and values in the {@link Map}.
+   */
+  private static Headers createHeaders(Map<String, List<String>> headers) {
+    Headers.Builder builder = new Headers.Builder();
+    for (Map.Entry<String, List<String>> header : headers.entrySet()) {
+      if (header.getKey() == null || header.getValue() == null) {
+        continue;
+      }
+      String name = header.getKey().trim();
+      for (String value : header.getValue()) {
+        String trimmedValue = value.trim();
+        Internal.instance.addLenient(builder, name, trimmedValue);
+      }
+    }
+    return builder.build();
+  }
+
+  private static Map<String, List<String>> varyHeaders(URLConnection urlConnection,
+      Headers responseHeaders) {
+
+    if (OkHeaders.hasVaryAll(responseHeaders)) {
+      // "*" means that this will be treated as uncacheable anyway.
+      return Collections.emptyMap();
+    }
+    Set<String> varyFields = OkHeaders.varyFields(responseHeaders);
+    if (varyFields.isEmpty()) {
+      return Collections.emptyMap();
+    }
+
+    Headers.Builder result = new Headers.Builder();
+    for (String fieldName : varyFields) {
+      String fieldValue = urlConnection.getRequestProperty(fieldName);
+      if (fieldValue != null) {
+        result.add(fieldName, fieldValue);
+      }
+    }
+    return OkHeaders.toMultimap(result.build(), null);
```

Comment:
True. I was trying to avoid a ton of overloads, and make use of some of the validation / eliding / trimming, etc. that Headers provides. I have inlined some code instead.